datadog/dogstatsd/base.py

```diff
@@ -239,6 +254,34 @@ def __init__(
         Default: True.
         More on this: https://docs.datadoghq.com/developers/dogstatsd/?tab=kubernetes#origin-detection-over-udp
         :type origin_detection_enabled: boolean
+
+        :param socket_timeout: Set timeout for socket operations, in seconds. Optional.
+        If sets to zero, never wait if operation can not be completed immediately. If set to None, wait forever.
+        This option does not affect hostname resolution when using UDP.
+        :type socket_timeout: float
+
+        :param telemetry_socket_timeout: Set timeout for the telemetry socket operations. Optional.
+        Effective only if either telemetry_host or telemetry_socket_path are set.
+        If sets to zero, never wait if operation can not be completed immediately. If set to None, wait forever.
+        This option does not affect hostname resolution when using UDP.
+        :type socket_timeout: float
+
+        :param disable_sender: Use a background thread to communicate with the dogstatsd server. Optional.
+        When enabled, a background thread will be used to send metric payloads to the Agent.
+        Default: True.
+        :type disable_sender: boolean
```

Comment:
Would that make sense to have something like `disable_background_sender` or similar, explicitly displaying that the mentioned sender is running in the background/in a thread, not just the normal one?
This could also be thought the other way around: `enable_sender_thread` or similar.

It might be over-thinking it, please discard if you think so.